Source/ARTRealtime.m

```diff
@@ -536,13 +569,26 @@ - (void)onClosed {
     }
 }
 
+- (void)onAuth {
+    [self.logger info:@"R:%p server has requested an authorise", self];
+    switch (self.connection.state) {
+        case ARTRealtimeConnecting:
+        case ARTRealtimeConnected:
+            [self transportReconnectWithRenewedToken];
+            break;
+        default:
+            NSAssert(false, @"Invalid Realtime state: expected Connecting or Connected");
+            break;
+    }
+}
+
 - (void)onError:(ARTProtocolMessage *)message {
-    // TODO work out which states this can be received in
     if (message.channel) {
         [self onChannelMessage:message];
     } else {
         ARTErrorInfo *error = message.error;
         if ([self shouldRenewToken:&error]) {
+            [self.transport close];
```

Comment:
I can't precise the problem that was occurring if the `close` wasn't called explicitly but the main reason I added the `close` call was because the `authorize` from `transportReconnectWithRenewedToken` takes time and the client shouldn't use the current token.